private.tf

```diff
@@ -1,10 +1,10 @@
 locals {
   private_count       = "${var.enabled == "true" && var.type == "private" ? length(var.availability_zones) : 0}"
-  private_route_count = "${var.enabled == "true" && var.type == "private" ? length(var.az_ngw_count) : 0}"
+  private_route_count = "${var.enabled == "true" && var.type == "private" && var.nat_gateway_enabled == "true" ? length(var.az_ngw_count) : 0}"
```

Comment:
length(**var.az_ngw_ids**)

Reply:
Let's use `var.az_ngw_count` here.
@mike-zipit please update.
`length(var.az_ngw_ids)` was causing TF errors before when this module was used with other modules (since TF can't calculate dynamic counts of resources that have not been created yet in external modules).

thanks @mderrier